pages_project.go

```diff
@@ -30,11 +31,15 @@ type PagesProjectSource struct {
 
 // PagesProjectSourceConfig represents the properties use to configure a Pages project source.
 type PagesProjectSourceConfig struct {
-	Owner              string `json:"owner"`
-	RepoName           string `json:"repo_name"`
-	ProductionBranch   string `json:"production_branch"`
-	PRCommentsEnabled  bool   `json:"pr_comments_enabled"`
-	DeploymentsEnabled bool   `json:"deployments_enabled"`
+	Owner                        string   `json:"owner"`
+	RepoName                     string   `json:"repo_name"`
+	ProductionBranch             string   `json:"production_branch"`
+	PRCommentsEnabled            bool     `json:"pr_comments_enabled"`
+	DeploymentsEnabled           bool     `json:"deployments_enabled"`
+	ProductionDeploymentsEnabled bool     `json:"production_deployments_enabled"`
+	PreviewDeploymentSetting     string   `json:"preview_deployment_setting"`
```

Comment:
This is an enum, can we have a proper type for it rather than allowing any value

Reply:
```suggestion
	PreviewDeploymentSetting     PreviewDeploymentSetting `json:"preview_deployment_setting"`
```